js/render/nodes/cube-sea-occlusion.js

```diff
@@ -0,0 +1,473 @@
+    } else {
+      // Used when we want to stress the GPU a bit more.
+      // Stolen with love from https://www.clicktorelease.com/code/codevember-2016/4/
+      return `
+      precision mediump float;
+
+      uniform sampler2D diffuse;
+      varying vec2 vTexCoord;
+      varying vec3 vLight;
+
+      vec2 dimensions = vec2(64, 64);
+      float seed = 0.42;
+
+      vec2 hash( vec2 p ) {
+        p=vec2(dot(p,vec2(127.1,311.7)),dot(p,vec2(269.5,183.3)));
+        return fract(sin(p)*18.5453);
+      }
+
+      vec3 hash3( vec2 p ) {
+          vec3 q = vec3( dot(p,vec2(127.1,311.7)),
+                 dot(p,vec2(269.5,183.3)),
+                 dot(p,vec2(419.2,371.9)) );
+        return fract(sin(q)*43758.5453);
+      }
+
+      float iqnoise( in vec2 x, float u, float v ) {
+        vec2 p = floor(x);
+        vec2 f = fract(x);
+        float k = 1.0+63.0*pow(1.0-v,4.0);
+        float va = 0.0;
+        float wt = 0.0;
+        for( int j=-2; j<=2; j++ )
+          for( int i=-2; i<=2; i++ ) {
+            vec2 g = vec2( float(i),float(j) );
+            vec3 o = hash3( p + g )*vec3(u,u,1.0);
+            vec2 r = g - f + o.xy;
+            float d = dot(r,r);
+            float ww = pow( 1.0-smoothstep(0.0,1.414,sqrt(d)), k );
+            va += o.z*ww;
+            wt += ww;
+          }
+        return va/wt;
+      }
+
+      // return distance, and cell id
+      vec2 voronoi( in vec2 x ) {
+        vec2 n = floor( x );
+        vec2 f = fract( x );
+        vec3 m = vec3( 8.0 );
+        for( int j=-1; j<=1; j++ )
+          for( int i=-1; i<=1; i++ ) {
+            vec2  g = vec2( float(i), float(j) );
+            vec2  o = hash( n + g );
+            vec2  r = g - f + (0.5+0.5*sin(seed+6.2831*o));
+            float d = dot( r, r );
+            if( d<m.x )
+              m = vec3( d, o );
+          }
+        return vec2( sqrt(m.x), m.y+m.z );
+      }
+
+      vec4 fragment_main() {
+        vec2 uv = ( vTexCoord );
+        uv *= vec2( 10., 10. );
+        uv += seed;
+        vec2 p = 0.5 - 0.5*sin( 0.*vec2(1.01,1.71) );
+
+        vec2 c = voronoi( uv );
+        vec3 col = vec3( c.y / 2. );
+
+        float f = iqnoise( 1. * uv + c.y, p.x, p.y );
+        col *= 1.0 + .25 * vec3( f );
+
+        return vec4(vLight, 1.0) * texture2D(diffuse, vTexCoord) * vec4( col, 1. );
+      }`;
+    }
+  }
+}
+
+export class CubeSeaNode extends Node {
+  constructor(options = {}) {
+    super();
+
+    // Test variables
+    // If true, use a very heavyweight shader to stress the GPU.
+    this.heavyGpu = !!options.heavyGpu;
+
+    // Number and size of the static cubes. Warning, large values
+    // don't render right due to overflow of the int16 indices.
+    this.cubeCount = options.cubeCount || (this.heavyGpu ? 12 : 10);
+    this.cubeScale = options.cubeScale || 1.0;
+
+    // Draw only half the world cubes. Helps test variable render cost
+    // when combined with heavyGpu.
+    this.halfOnly = !!options.halfOnly;
```

Comment:
Similarly to the heavy option, you could drop some of these testing options like `halfOnly` and `autoRotate`, which only really exist to provoke specific scenarios that some implementation tripped on. Not required, but might help the readability a bit.

Reply:
done